Make Box's Engine be randomised between SM, Tesla and Singulo V2 [TEST-MERGE]

r4d6 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This make the Engine in Engineering of Box not always the SM, also add a framework for more randomized area of the station.

I took most of the code from Yogstation

I ask for it to be test-merged just in case I fucked something up.
This is a cleaner version.

## Why It's Good For The Game

This make it so that it isn't always the SM the main engine and add the framework to be able to randomized different area of the station like Bar or Maintenance.

## Changelog
:cl:
add: Added submaps for the SM, Tesla and Singulo
add: Added a placeholder on Boxstation for the Engines
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

r4d6 committed:
Add Engines submaps

r4d6 committed:
Update Boxstation to have Engines landmark

r4d6 committed:
Adding a list for the stations rooms

r4d6 committed:
Fix a possible alert

r4d6 committed:
Add the stuff to seed the station with rooms

r4d6 committed:
Add the landmarks

r4d6 committed:
Add the file for making the submaps actual ruins

r4d6 committed:
Merge pull request #1 from r4d6/master

Wrong branch

r4d6 committed:
Add the sprite for the landmark

r4d6 committed:
Make it include my files

Commit:
Merge remote-tracking branch 'upstream/master' into substationV2

Commit:
Redo my changes on Box

Hopefully this will fix the conflict with the fire alarms.

Commit:
QoL Update

Move the Engines to a subfolder to be cleaner when there will be 20 submaps on different maps and change the Tesla Submap to incorporate a few changes I did for the Singulo submap.

Commit:
Make things even more organized

Ghommie commented on code/controllers/subsystem/air.dm:
```diff
@@ -392,6 +392,9 @@ SUBSYSTEM_DEF(air)
 		CHECK_TICK
 
+	if(!initialized)
+		return
```
Move these two lines at the start of the proc.

Ghommie commented on code/controllers/subsystem/mapping.dm:
```diff
@@ -524,4 +527,17 @@ GLOBAL_LIST_EMPTY(the_station_areas)
+	var/list/station_room_templates = list()
+
+/datum/controller/subsystem/mapping/proc/seedStation()
+	message_admins("Seeding Station")
```
Remove this local testing line.

Ghommie commented on code/controllers/subsystem/mapping.dm:
```diff
@@ -524,4 +527,17 @@ GLOBAL_LIST_EMPTY(the_station_areas)
+	for(var/V in GLOB.stationroom_landmarks)
+		var/obj/effect/landmark/stationroom/LM = V
+		LM.load()
+	message_admins("Station seeded")
```
Same here.

Ghommie commented on code/game/objects/effects/landmarks.dm:
```diff
@@ -432,3 +432,48 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/start/new_player)
+	var/datum/map_template/template = SSmapping.station_room_templates[template_name]
+	if(!template)
+		return FALSE
+	testing("Ruin \"[template_name]\" placed at ([T.x], [T.y], [T.z])")
```
```suggestion
	testing("Room \"[template_name]\" placed at ([T.x], [T.y], [T.z])")
```
Because it's a station room, not a ruin... nothing important.

r4d6 replied:
Funny thing is, I never saw this line. Though I don't really know where to see it either.

Ghommie commented on code/game/objects/effects/landmarks.dm:
```diff
@@ -432,3 +432,48 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/start/new_player)
+	template.loaded++
+	GLOB.stationroom_landmarks -= src
+	qdel(src)
+	message_admins("Stationroom loaded successfully")
```
Remove this testing line.

Ghommie reviewed (approved):
Apply the suggestions below and it should be ready for merge.
P.S. Travis errored on its own. I had to help the author in several occasions and can assume the port should be fine now.

Commit:
Suggestions

Commit:
Fix indentation

Commit:
QoL Update

- Rename the cameras and place two more

- Unanchor the shields generators and emitters because they were bugged and will save more time

- Put more rad closets on the Singulo

- Put more insuls on the Tesla since they protect

- Moved the Engineer clothing locker to Boxstation instead of having it in the Engine

- Added a button to the shutters on the north side of PA

- Added a marker to show where to place the console

- Rewire the power wire to be more in a straight line.

Commit:
Forgot a "

Commit:
Forgot a button on the Tesla

Commit:
Connect the fuel pipe to somewhere

Commit:
Fix areas and tiles getting ripped off

Commit:
Also give Tesla's emitters R-floors

silicons reviewed (dismissed):
make it possible to force a specific engine from config using its id tag please.

Commit:
Add the option to restrict the Engine that spawn

Commit:
Revert "Add the option to restrict the Engine that spawn"

This reverts commit e3e1f9163538fb221a0697f6e36448032b3199d5.

Commit:
Revert "Revert "Add the option to restrict the Engine that spawn""

This reverts commit d851beea169841569078665b771b6f154700e26e.

Commit:
Add config options

Commit:
Move the config up to avoid conflict
